### PR TITLE
Internationalization and English Localization

### PR DESCRIPTION
**Pull request by YaxelPerez**
This PR resolves issue #28

### CODE REVIEW

**Comment by kfogel** [comment]
The commit message needs to mention the issue number too.

Remember, the commit messages will show up in the Git log history, whereas comments that are only in the PR ticket in the GitHub web site will not.

File: TorqueDataConnect/include/TorqueDataConnectConfig.php [hunk 1]
```diff
@@ -5,7 +5,7 @@ class TorqueDataConnectConfig {
   public static function convertPageToColumnConfig($page) {
     $title = Title::newFromText($page);
     if(!$title->exists()) {
-      array_push(self::$errors, $page . " does not exist, but is reference in config as a column config.");
+      array_push(self::$errors, wfMessage("convert-column-page-ne", $page)->parse());
```
Comment (kfogel) [inline TorqueDataConnect/include/TorqueDataConnectConfig.php:8]: This is more a question for the next reviewer (presumably @frankduncan) than it is a comment -- I'm assuming that the next reviewer has both more i18n experience and more PHP experience than I have :-).

Is `wfMessage()` a built-in thing that we automatically get just by virtue of using PHP?  It automatically knows to look in `i18n/<two-letter-ISO-639-1-language-code>.json` for the strings for a given localization?

I think I came in expecting the change to be larger.  I thought there'd be an import of some i18n library or something -- but maybe that's not necessary if a perfectly fine one is built in already.  (I also sort of expected there to be more strings that would need replacing!  But @frankduncan is better positioned than I am to know how many strings need to be internationalized in Torque.)

**Comment by frankduncan** [comment]
> Do any of our plugins need to be i18n'd as well? I'm talking about [SimpleFavorites](https://github.com/OpenTechStrategies/SimpleFavorites), [PickSome](https://github.com/OpenTechStrategies/PickSome/), [ActivityLog](https://github.com/OpenTechStrategies/ActivityLog), and [TeamComments](https://github.com/OpenTechStrategies/TeamComments).

Nope!  I checked those.  There is one issue that I filed with those for a small change assigned to Yaxel, but otherwise we should be good :)

File: TorqueDataConnect/include/TorqueDataConnectConfig.php (hunk 1 above)
Comment (frankduncan) [reply TorqueDataConnect/include/TorqueDataConnectConfig.php:8]: `wfMessage()` is one of the MediaWiki i18n functions, and yes, it knows how to look in all the right places as configured by your plugin in extension.json

File: TorqueDataConnect/include/hooks/TorqueDataConnectHooks.php [hunk 2]
```diff
@@ -73,7 +73,7 @@ public static function siteNoticeAfter(&$siteNotice, $skin) {
     if(sizeof($configErrors) > 0) {
       $html = "";
       $html .= "<div style='border:2px solid #AA3333;padding:10px;text-align:left;margin-top:10px;background-color:#FFEEEE'>";
-      $html .= "<h2 style='margin-top:0px;border-bottom:0px'>Torque Data Configuration Alert!</h2>";
+      $html .= "<h2 style='margin-top:0px;border-bottom:0px'>" . wfMessage("data-config-alert")->plain() . "</h2>";
```
Comment (frankduncan) [inline TorqueDataConnect/include/hooks/TorqueDataConnectHooks.php:76]: I don't see this in the en.json changes.

**Comment by frankduncan** [comment]
As an aside, @YaxelPerez, if you find any of the language makes it really challenging to localize to spanish, please change!

**Comment by YaxelPerez** [comment]
@frankduncan 
> the plugin isn't showing up correctly in Special:Version and Special:SpecialPages

working on this as of today (jun 12). I'm assuming it's the same issue as [Picksome Issue 1](https://github.com/OpenTechStrategies/PickSome/issues/1)

from what I can tell, there shouldn't be any difficulty w/ internationalization to Spanish

**Comment by YaxelPerez** [comment]
Resuming again Jun 17. [MediaWiki documentation](https://www.mediawiki.org/wiki/Manual:Special_pages#The_aliases_file) on special page naming is likely outdated. Trying to work off of SimpleFavorites repo. This same issue is present in [PickSome](https://github.com/OpenTechStrategies/PickSome/issues/1) and the solution will carry over.

**Comment by kfogel** [comment]
*nod*  Thanks for the update, @YaxelPerez.  I'll continue watching this space!

**Comment by YaxelPerez** [comment]
likely finishing this off mon, jun 22. will edit this comment with more details

**Comment by YaxelPerez** [comment]
Screen Shot 2020-06-23 at 12 46 03 PM

Getting the desired behavior now.

File: TorqueDataConnect/extension.json [hunk 3]
```diff
@@ -43,7 +43,8 @@
     "torquedataconnectuploadattachment": "TorqueDataConnectUploadAttachment"
   },
   "ExtensionMessagesFiles": {
-		"TorqueDataConnectMagic": "TorqueDataConnect.i18n.php"
+		"TorqueDataConnectMagic": "TorqueDataConnect.i18n.php",
```
Comment (frankduncan) [inline TorqueDataConnect/extension.json:46]: Looks like this line replaces spaces with a tab character.  Can you change it back to use spaces?

Comment (kfogel) [reply TorqueDataConnect/extension.json:46]: +1  (see also related discussion in [ActivityLog issue #3](https://github.com/OpenTechStrategies/ActivityLog/issues/3), which refers to a probably-soon-to-be-in-effect `CONTRIBUTING.md` file that documents this convention of spaces-not-tabs).

**Comment by kfogel** [comment]
@YaxelPerez Did you see @frankduncan's request above?

**Comment by frankduncan** [comment]
This looks good @YaxelPerez I'm merging so that I can get some changes in on top of yours, sorry!

**Pull request merged**